prow/gcsupload/options.go

```diff
@@ -66,17 +66,8 @@ type Options struct {
 // Validate ensures that the set of options are
 // self-consistent and valid.
 func (o *Options) Validate() error {
-	for _, extension := range o.Extensions.Strings() {
-		parts := strings.SplitN(extension, ":", 2)
-		if len(parts) != 2 {
-			return fmt.Errorf("invalid extension %q: missing colon delimiter", extension)
-		}
-
-		_, mediaType := parts[0], parts[1]
-		_, _, err := mime.ParseMediaType(mediaType)
-		if err != nil {
-			return fmt.Errorf("invalid media type %q in extension %q: %v", mediaType, extension, err)
-		}
+	if len(o.mediaTypes.Strings()) > 0 {
+		return fmt.Errorf("the mediaTypes property is only for command-line support; use GCSConfiguration.MediaTypes instead")
```

Comment:
I think we can assume someone calls `Complete()` and we don't need to set the value to the zero value there and check it here

Reply:
> I think we can assume someone calls `Complete()` and we don't need to set the value to the zero value there and check it here

I've dropped the check.  Kept the zero, because it's one-liner and ensures folks won't attempt to consume `Options.mediaTypes`.  But I can drop that too if you need me to ;).